sonar-db/src/main/java/org/sonar/db/issue/IssueChangeDao.java

```diff
@@ -89,18 +90,23 @@ public DefaultIssueComment selectCommentByKey(String commentKey) {
     }
   }
 
-  List<IssueChangeDto> selectByIssuesAndType(DbSession session, Collection<String> issueKeys, String changeType) {
-    if (issueKeys.isEmpty()) {
-      return Collections.emptyList();
+  public List<IssueChangeDto> selectByTypeAndIssueKeys(DbSession session, Collection<String> issueKeys, String changeType) {
+    return DatabaseUtils.executeLargeInputs(issueKeys, new SelectByIssueKeys(session.getMapper(IssueChangeMapper.class), changeType));
+  }
+
+  private static class SelectByIssueKeys implements Function<List<String>, List<IssueChangeDto>> {
```

Comment:
No, this function is specific to this method and isn't/won't be reused elsewhere.